test/python/circuit/test_singleton_gate.py

```diff
@@ -251,3 +253,26 @@ def test_positional_label(self):
         label_gate = SXGate("I am a little label")
         self.assertIsNot(gate, label_gate)
         self.assertEqual(label_gate.label, "I am a little label")
+
+    def test_immutable_pickle(self):
+        gate = SXGate()
+        self.assertFalse(gate.mutable)
+        with io.BytesIO() as fd:
+            pickle.dump(gate, fd)
+            fd.seek(0)
+            copied = pickle.load(fd)
+        self.assertFalse(copied.mutable)
+
+    def test_mutable_pickle(self):
```

Comment:
One thing that surprised me with this issue is that it was not caught in this repo before the singleton gate PR was merged. I expect there is a good set of tests for transpiling conditional gates, but likely the tests all transpile a single circuit and don't hit the pickling issues that come up with parallel transpilation. I wonder if the tests added here are enough or if there should be more tests of pickling and unpickling circuits.

Reply:
There's actually not a _vast_ amount of testing of conditional gates (with `.condition` set), because they typically weren't executable on anything but simulators until IBM hardware grew support for midcircuit measurements and more generalised control flow, at which point we began pivoting to `IfElseOp`.

The bug getting through points to a deficiency in integration testing around conditional gates for sure, and we can add more tests along those lines until `Instruction.condition` is completely removed in 1.0.  Conditionals after transpilation are also harder to write good tests about, because we can't just test for unitary equivalence or anything like that when they're present, so it's easier for stuff to slip through.